File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/di/main/MainModule.java
package pl.tobzzo.miechowskiepowietrze.di.main;

import dagger.Module;
import dagger.Provides;
import pl.tobzzo.miechowskiepowietrze.mvp.main.MainContract;
import pl.tobzzo.miechowskiepowietrze.mvp.main.MainPresenter;

@Module
public class MainModule {
  @Provides MainPresenter provideMainPresenter(MainContract.MainView mainView) {
    return new MainPresenter(mainView);
  }
}

File: app/src/main/java/pl/tobzzo/miechowskiepowietrze/di/MainModule.kt
package pl.tobzzo.miechowskiepowietrze.di


import dagger.Binds
import dagger.Module
import pl.tobzzo.miechowskiepowietrze.example.ExampleInterface
import pl.tobzzo.miechowskiepowietrze.example.MpowExampleImplementation
import pl.tobzzo.miechowskiepowietrze.network.MpowNetworkComponent
import pl.tobzzo.miechowskiepowietrze.network.NetworkComponent

@Module
abstract class MainModule {

  @Binds
  internal abstract fun bindExample(example: MpowExampleImplementation): ExampleInterface
}
